### PR TITLE
Implement Middleware

### PR DESCRIPTION
**Pull request by vitalikdanchenko**

### CODE REVIEW

File: lib/talkable.rb [hunk 1]
```diff
@@ -13,6 +17,18 @@ def configure(config = nil)
     def configuration
       @configuration ||= Talkable::Configuration.new
     end
+
+    def with_uuid(uuid)
```
Comment (Tonkonozhenko) [inline lib/talkable.rb:27]: I suppose that it can share `@visitor_uuid` between requests. It's not OK

Comment (vitalikdanchenko) [inline lib/talkable.rb:27]: @Tonkonozhenko the main idea is in it, what exactly is not okay?

Comment (bogdan) [inline lib/talkable.rb:27]: See the implementation of similar method in `Time.use_zone` to ensure correct implementation

File: talkable.gemspec [hunk 2]
```diff
@@ -17,6 +17,12 @@ Gem::Specification.new do |s|
 
   s.required_ruby_version = ">= 1.9.3"
 
+  if RUBY_VERSION < '2.2.2'
```
Comment (Tonkonozhenko) [inline talkable.gemspec]: You should add comparison using `Gem::Version`

``` ruby
v1, v2 = '2.2.2', '2.2.10'
=> ["2.2.2", "2.2.10"]
[v1 < v2, Gem::Version.new(v1) < Gem::Version.new(v2)]
=> [false, true]
```

**Comment by Tonkonozhenko** [comment]
LGTM

**Pull request merged**